capstone/src/test/java/com/google/sps/servlets/student/StudentServletTest.java

```diff
@@ -254,9 +255,11 @@ public void doPost_studentClicksJoinClub() throws ServletException, IOException
     Query query = new Query(MEGHA_EMAIL);
     PreparedQuery results = datastore.prepare(query);
     Entity student = ImmutableList.copyOf(results.asIterable()).get(0);
-    String clubList = student.getProperty(Constants.PROPERTY_CLUBS).toString();
+    ImmutableList<String> clubList =
+        ImmutableList.copyOf((ArrayList<String>) student.getProperty(Constants.PROPERTY_CLUBS));
 
-    Assert.assertEquals(ImmutableList.of(CLUB_2).toString(), clubList);
+    Assert.assertEquals(1, clubList.size());
+    Assert.assertEquals(CLUB_2, clubList.get(0));
```

Comment:
Another thing you could check is if the response is redirecting to the correct place using `Mockito.verify(response).sendRedirect("link")` since this is also part of what doPost/doGet methods might do. Up to you!

Reply:
Done! Thank for bringing this up :)